### PR TITLE
fix(c/sql): stored procedure/function metadata

### PR DESCRIPTION
**Pull request by zregvart**
When fetching metadata for functions in PostgreSQL we need to use stored
procedure metadata, not function metadata as it returns nothing.

Ref. https://issues.redhat.com/browse/ENTESB-16358

fix(c/sql): make sure test tear down is done (766e285)

We need to make sure that any object we create is dropped in the test.
Otherwise we might get flaky tests that depend on the order of
execution.

(cherry picked from commit fbcbfeac2a2111e7f43f60458b88a7472bff5778)

### CODE REVIEW

**Comment by zregvart** [comment]
retest this please

**Comment by avano** [comment]
retest this please

File: app/connector/sql/src/main/java/io/syndesis/connector/sql/SqlSupport.java [hunk 1]
```diff
@@ -49,15 +47,13 @@ public static Connection createConnection(final Map<String, Object> properties)
     }
 
     public static StoredProcedureMetadata getFunctionOrProcedureMetadata(final Connection connection, final String catalog,
```
Comment (claudio4j) [inline app/connector/sql/src/main/java/io/syndesis/connector/sql/SqlSupport.java]: As there is no fetchFunctionColumns call, renaming the method back to getProcedureMetadata and adding a comment about why not using fetchFunctionColumns.

**Review by claudio4j — CHANGES_REQUESTED** [review]

**Comment by zregvart** [comment]
Thanks for the review, @claudio4j can you have another look?

**Review by claudio4j — APPROVED** [review]

**Pull request merged**